### PR TITLE
ECDC-2940 serializer flushing, EV44 Serializer, and split monitors

### PR DESCRIPTION
**Pull request by JennyLouise**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: src/modules/ttlmonitor/TTLMonitorInstrument.cpp [hunk 1]
```diff
@@ -49,9 +50,14 @@ void TTLMonitorInstrument::processMonitorReadouts(void) {
   // possible, but rings and fens
   // could still be outside the configured range, also
   // illegal time intervals can be detected here
-  assert(Serializer != nullptr);
-  Serializer->pulseTime(ESSReadoutParser.Packet.Time
+
+  
+  //TODO, have proper assertion heres
+  //assert(Serializers != nullptr);
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/TTLMonitorInstrument.cpp:56]: remove

File: src/modules/ttlmonitor/TTLMonitorInstrument.cpp [hunk 2]
```diff
@@ -49,9 +50,14 @@ void TTLMonitorInstrument::processMonitorReadouts(void) {
...
+  for (EV44Serializer &Serializer : Serializers){
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/TTLMonitorInstrument.cpp:57]: store counters inside serializer - count different kinds of flushes
store timer inside serializer

File: src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp [hunk 3]
```diff
@@ -117,21 +119,21 @@ class TTLMonitorInstrumentTest : public TestBase {
...
-    serializer = new EV42Serializer(115000, "ttlmonitor");
+    serializers.push_back(EV44Serializer(115000, "ttlmonitor"));
     counters = {};
 
     memset(&PacketHeader, 0, sizeof(PacketHeader));
 
-    ttlmonitor = new TTLMonitorInstrument(counters, ModuleSettings, serializer);
-    ttlmonitor->setSerializer(serializer);
+    ttlmonitor = new TTLMonitorInstrument(counters, ModuleSettings, serializers);
+    // ttlmonitor->setSerializer(serializers);
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp:136]: remove

File: src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp [hunk 4]
```diff
@@ -186,8 +188,8 @@ TEST_F(TTLMonitorInstrumentTest, BeamMonitorTOF) {
 /// THIS IS NOT A TEST, just ensure we also try dumping to hdf5
 TEST_F(TTLMonitorInstrumentTest, DumpTofile) {
   ModuleSettings.FilePrefix = "deleteme_";
-  TTLMonitorInstrument TTLMonDump(counters, ModuleSettings, serializer);
-  TTLMonDump.setSerializer(serializer);
+  TTLMonitorInstrument TTLMonDump(counters, ModuleSettings, serializers);
+  //TTLMonDump.setSerializer(serializer);
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp:192]: remove

**Review by mortenjc — APPROVED** [review]

**Pull request merged**